central/delegatedregistryconfig/service/service.go

```diff
@@ -0,0 +1,213 @@
+	config, exists, err := s.dataStore.GetConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving config %w", err)
+	}
+
+	if !exists {
+		return &v1.DelegatedRegistryConfig{}, nil
+	}
+
+	return convert.StorageToAPI(config), nil
+}
+
+// GetClusters returns the list of clusters (id + name) that are eligible for delegating scanning
+// requests (ie: only clusters with scanners that understand the delegated registry config)
+func (s *serviceImpl) GetClusters(ctx context.Context, _ *v1.Empty) (*v1.DelegatedRegistryClustersResponse, error) {
+	clusters, err := s.getClusters(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving clusters %w", err)
+	}
+
+	if len(clusters) == 0 {
+		return nil, status.Error(codes.NotFound, "no valid clusters found")
+	}
+
+	return &v1.DelegatedRegistryClustersResponse{
+		Clusters: clusters,
+	}, nil
+}
+
+// PutConfig updates Central's delegated registry config
+func (s *serviceImpl) PutConfig(ctx context.Context, config *v1.DelegatedRegistryConfig) (*v1.DelegatedRegistryConfig, error) {
+	if err := s.validate(ctx, config); err != nil {
+		return nil, fmt.Errorf("%w: %v", errox.InvalidArgs, err.Error())
+	}
+
+	if err := s.dataStore.UpsertConfig(ctx, convert.APIToStorage(config)); err != nil {
+		return nil, fmt.Errorf("upserting config %w", err)
+	}
+
+	return config, nil
+}
+
+func (s *serviceImpl) validate(ctx context.Context, config *v1.DelegatedRegistryConfig) error {
+	if config == nil {
+		// this block not reachable via GRPC-gateway invocations
+		return errors.New("config missing")
+	}
+
+	if config.EnabledFor == v1.DelegatedRegistryConfig_NONE {
+		// ignore rest of config
+		return nil
+	}
+
+	// validate the default cluster id
+	if config.DefaultClusterId == "" {
+		return errors.New("default cluster id required if enabled is not NONE")
+	}
+
+	// get the valid clusters
+	validClusters, err := s.getValidClusterIds(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to validate config %w", err)
+	}
+
+	var errorList []error
+	if !validClusters.Contains(config.DefaultClusterId) {
+		errorList = append(errorList, fmt.Errorf("default cluster %q is not valid", config.DefaultClusterId))
+	}
+
+	// validate the registries / clusters
+	for _, r := range config.Registries {
+
+		// if a cluster id was provided, check if its valid
+		if r.ClusterId != "" && !validClusters.Contains(r.ClusterId) {
+			errorList = append(errorList, fmt.Errorf("cluster %q is not valid", r.ClusterId))
+		}
+
+		if r.RegistryPath == "" {
+			errorList = append(errorList, errors.New("missing registry path"))
+		}
+	}
+
+	return errors.Join(errorList...)
+}
+
+// getClusters returns all clusters annotated with a flag indicating if cluster is valid
+// for use as a delegation target. All clusters are returned instead of just valid clusters
+// so that a consumer (ie: the UI) can show the friendly name of clusters that may no longer
```

Comment:
I think this part may be a little confusing, and I think you can just remove the explanation to resolve that.

Reply:
I think its important to explain the why the valid flag is there if its questioned in the future. Perhaps I can re-word it to make it more clear